src/parser/mod.rs

```diff
@@ -7122,7 +7122,17 @@ impl<'a> Parser<'a> {
         } else {
             None
         };
-        Ok(ViewColumnDef { name, options })
+        let data_type = if dialect_of!(self is ClickHouseDialect) {
+            let (data_type, _) = self.parse_data_type_helper()?;
```

Comment:
```suggestion
            let (data_type, _) = self.parse_data_type()?;
```
should this be parse_data_type instead?